src/qute/commands/registerCommands.ts

```diff
@@ -91,17 +165,44 @@ export function registerConfigurationUpdateCommand(context: ExtensionContext) {
         config.workspaceConfiguration.update(section, value, config.target);
         break;
       }
+      case ConfigurationItemEditType.Remove: {
+        removeFromPreferenceArray(config, section, value);
+        break;
+      }
     }
   }));
 }
 
-function getSettingsValue(value: any, section: string) {
+/**
+   * Sets the `editorLangIdSupportsQute` context to `true` if the editor language id
+   * is a qute template. Sets to `false` otherwise.
+   */
+export async function updateQuteContext(document: TextDocument) {
+  const quteSupported = document.languageId.includes('qute');
+  await commands.executeCommand('setContext', 'editorLangIdSupportsQute', quteSupported);
+  if (quteSupported) {
+    await checkQuteValidationFromExclusionContext(document.uri);
+  }
+}
+
+/**
+   * Sets the `editorQuteValidationEnabled` based on `qute.command.validation.template.status` and `qute.validation.enabled`.
+   */
+async function checkQuteValidationFromExclusionContext(uri: Uri) {
+  const templateUri = uri.toString();
+  const result: TemplateValidationStatus = await commands.executeCommand(QuteClientCommandConstants.EXECUTE_WORKSPACE_COMMAND, QuteServerCommandConstants.QUTE_VALIDATION_TEMPLATE_STATUS, templateUri);
+  await commands.executeCommand('setContext', 'editorQuteValidationEnabled', (result.validationEnabled && (!result.excluded || result.excluded.length === 0)));
+}
+
+function getSettingsValue(value: any, section: string, editType: ConfigurationItemEditType) {
   switch (section) {
     case QuteSettings.QUTE_VALIDATION_EXCLUDED:
-      return workspace.asRelativePath(Uri.parse(value), false);
-    default:
-      return value;
+      if (editType === ConfigurationItemEditType.Add) {
+        return workspace.asRelativePath(Uri.parse(value), false);
```

Comment:
This logic seems to add an extra path separator(e.g. `/src...` instead of `src...` ) at the beginning on occasion, and so I feel this settings value retrieval introduces more problems then it fixes.

Reply:
Using workspace.asRelativePath(Uri.parse(value), false) seems to add more problems than just using the raw template uri, since the template uri is already relative to the workspace. I suggest we dont use getSettingsValue simply because it introduces this problem.

Reply:
I believe we can achieve the same result by calling path.relative(workspace.workspaceFolders[0].uri.toString(), window.activeTextEditor.document.uri.toString()) since we're looking at the activeTextEditor

Reply:
As I said this code avoids storing template uri with file:// scheme which is not portable. Try to remove getseetingsvalue and you will see what I mean.

Im surprised that you have some trouble (which trouble?) because this code has been tested before by @fbricon and me and it works good.

Could you explain what is your problem please?

Reply:
The toggle command is visible from the command palette, however, when we trigger that command, the uri doesnt exist. To fix this, I added an undefined check to use the path relative to the editor file and the workspace folder. However, when this uri is appended to the configuration edit, getSettingsValue adds a leading `/`. Can we avoid this and just find the relative path in the command listener instead?

Reply:
i.e. my fix in toggle on/off:
```
    const templateUri = uri ? uri.toString() : path.relative(workspace.workspaceFolders[0].uri.toString(), window.activeTextEditor.document.uri.toString());

```

Reply:
Please keep the existing coďe. Your code doesnt tzke care of multiple workspace folders.

Your problem is when you execute command with palette if I understand.

To fix that you should take the document uri of the current text editor.

Reply:
Did you try the most recent PR ? I don't see the commands in the command palette. The command palette contribution of the commands with `"when": "false"` seems to work.

Reply:
Nice! @AlexXuChen is it good now to keep the existing code with workspace.asRelativePath(Uri.parse(value), false) ? Or do you have another usecase which causes problems ?

Reply:
If the user can't use the command palette commands, then yser everything is working as intended